packages/element/src/serialize.js

```diff
@@ -54,6 +56,20 @@ import RawHTML from './raw-html';
 
 const { Provider, Consumer } = createContext();
 
+const Dispatcher = {
+	readContext: noop,
+	useContext: noop,
+	useMemo: noop,
+	useReducer: () => [],
+	useRef: () => ( { current: '' } ),
+	useState: () => [],
+	useLayoutEffect: noop,
+	useCallback: noop,
+	useImperativeHandle: noop,
+	useEffect: noop,
+	useDebugValue: noop,
+};
+
```

Comment:
This was mostly just to get the ball rolling on one option which is monkey patching the `ReactCurrentDispatcher` before doing the render.  However, if we go this route, we'll probably have to reproduce similar logic for hooks [found here](https://github.com/facebook/react/blob/master/packages/react-dom/src/server/ReactPartialRendererHooks.js).  This seems fragile and increases maintenance burden.  The other (maybe not significant) potential issue is that essentially any login in a `useEffect` or `useLayout` hook in a rendered component will not fire if we use something similar to `ReactPartialRendererHooks` because this serializer is effectively doing similar to what server type tasks do (as opposed to browser).

If `renderToString` is mostly used for the `save` function of blocks (stateless components) I wonder if we should simply detect usages of `withSelect`/hooks and simply NOT support them (throw errors?).